src/databricks/labs/lsql/backends.py

```diff
@@ -200,7 +200,7 @@ def execute(self, sql: str) -> None:
     def fetch(self, sql: str) -> Iterator[Row]:
         logger.debug(f"[spark][fetch] {self._only_n_bytes(sql, self._debug_truncate_bytes)}")
         try:
-            return self._spark.sql(sql).collect()
+            return iter(self._spark.sql(sql).collect())
```

Comment:
Why do we need to do this, by the way? Shouldn't we just change the result type to Iterable?

Reply:
It's just make it consistent with the current return type of `fetch` defined in SqlBackend, StatementExecutionBackend and MockBackend. 
Instead, we can also change the result type to Iterable in all these places. But it may break downstream code who is assuming it returns Iterator, and is using method like `next()`